### ISSUE TITLE
Run product-market-fit-indicator and jobs-to-be-done user research campaign

### ISSUE THREAD

**Issue by yourcontact** [getferdi/ferdi#663]
~**Is your feature request related to a problem? Please describe.**~
~A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]~

**This issue is about product marketing and user research**

I would like to help better understand how we can position Ferdi as a product on the market; how we can better define what Ferdi is.

It seems that the current description of Ferdi does not match what people use it for and the value the product offers. For instance, one of those descriptions is a "messaging browser." However, Ferdi aggregates a lot more applications than messaging applications (e.g. Airtable, GitHub, etc). That is for messaging -- for browsing, I feel Ferdi offers a lot more benefits than just browsing.

A few examples:
- we talk a lot about [workspace(s)](https://github.com/getferdi/ferdi/issues?q=is%3Aissue+is%3Aopen+workspace);  more than [messaging](https://github.com/getferdi/ferdi/issues?q=is%3Aissue+is%3Aopen+messaging);
- some talk about [control over the apps they use](https://twitter.com/kytwb/status/1254320029711175682), maybe there is some unrealised value offering sanity and peace of mind to users amid the jungle of apps and notifications;  
- some are [confused about what Ferdi is](https://github.com/getferdi/ferdi/issues/581#issue-600827546). 

I believe we can improve the messaging and positioning of Ferdi so people understand what benefits Ferdi bring.

This work would aslo be a reward for contributors to know that their work is promoted at best in the market.

**Describe the solution you'd like**
(A clear and concise description of what you want to happen).

The solution is to run a survey and ask the following questions to users: 

- "How would you feel if you could no longer use Ferdi?" That is to find out [Product Market Fit Indicator](https://www.startup-marketing.com/the-startup-pyramid/) (multiple choice);
  - "very disappointed"
  - "somewhat disappointed"
  - "not disapointed"
- "Why did you download Ferdi in the first place?" (open question) That is to find out [what "jobs" people "hire" Ferdi for](https://www.youtube.com/watch?v=sfGtw2C95Ms);
- "What is the main benefit you receive from Ferdi?"  (open question)  That is to help us define the value proposition, messaging and positioning;
- "How can we improve Ferdi for you?"  (open question)  That is to feed product development vision.

We could run the survey in-app using a Typeform link: 
https://rp28.typeform.com/to/E3phJT
And feed the results to a Google Spreadsheet or an Airtable accessible by all.

Then I would analyse the answers and share the hightlights in this issue. The deliverables would be: 
- product market fit indicator figure ([more on product market fit indicator](https://www.startup-marketing.com/the-startup-pyramid/));
-  "jobs" people "hire" Ferdi for ([more on jobs-to-be-done](https://www.youtube.com/watch?v=sfGtw2C95Ms));
- a list of categorised answers of the main benefit users receive from Ferdi;
- a list of categorised answers of the improvement users want.

Let me know what you think.
Let me know if you have suggestions to make the process more transparent.

~**Describe alternatives you've considered**~
~A clear and concise description of any alternative solutions or features you've considered.~

~**Additional context**~
~Add any other context or screenshots about the feature request here.~

**Comment by vantezzen** [comment]
This sounds amazing, thank you for offering your help on this!
How would you suggest we inform users about such a survey?

**Comment by yourcontact** [comment]
Hey @vantezzen, thank you for asking. I was about to jump back to this issue to add details about the implementation.

So -- the way I did before (for mobile apps) was to have a pop-up in the app with a message/copy saying "We would love to have your feedback" -- or something along this line. Then a user would tap on the pop-up and be redirected to Typeform in a webview. I don't think we want to have the user take the survey in a webview here in Ferdi. I am sharing my past experience FYI. That's said, that would be fantastic if we can have a pop-up somewhere in the app to maximize the number of responses we collect.

Based on the number of users Ferdi has, we should aim at collecting 300-350 responses for the results to be [statistically significant ](https://en.wikipedia.org/wiki/Statistical_significance) (roughly, that means so we can trust the results to be true). Less responses could work too; the sample size guidance of 300-350 responses is if we want to run that survey by the book, statistically speaking.

Also, the sample of respondents should be random enough, that is why I suggest have the link in the app (to survey all kinds of users), not just on Github where only a specific sample of Ferdi's users come.

If we decide to go for an in-app pop-up with a link to [the survey on Typeform](https://rp28.typeform.com/to/E3phJT), we leave that pop-up until we get 350 answers.

Let me know if you want me to clarify, or have questions.

**The stack**

I would like to be as transparent as I can in collecting data from users. We ask those 4 questions; no other data is collected on users.

I would be using my Typeform account then download the result to LibreOffice Calc, and share the spreadsheet (with all results) and my analysis here, in this issue. Let me know if you think I should do otherwise.

**Comment by kytwb** [comment]
Thank you for sharing your experience with us @yourcontact! I find the idea very interesting! What do you think about leveraging the "Support Ferdi" screen of the app (via https://github.com/getferdi/ferdi/issues/651) to link to the Typeform? I think a pop up (new window) could work; without an address bar and properly sized, it would combine well with the minimalistic UI/effective UX of Typeform.

**Comment by yourcontact** [comment]
> What do you think about leveraging the "Support Ferdi" screen of the app (via #651) to link to the Typeform?

We could have the link to Typeform there but I am not sure how many click-through, and responses, we would get. I am afraid the volume would be too low. Realistically, we need to collect 300-350 answers over a period of a few days. If you think that could work, we can try to have a link/button to Typeform on that screen.

>  I think a pop up (new window) could work; without an address bar and properly sized, it would combine well with the minimalistic UI/effective UX of Typeform.

Oh okay, are we talking about a webview here? The way you describe would work well. The user could take the survey, or click on one of his services icon to discard the survey. 

So basically we would have the Typeform screen with the survey show up, let's say when a user starts Ferdi? Like this:

<img width="1084" alt="Screenshot 2020-04-27 at 23 24 21" src="https://user-images.githubusercontent.com/46404814/80422370-521d7180-88de-11ea-90ce-0aa1463f3d3c.png">

**Comment by kytwb** [comment]
@yourcontact We originally forked Franz to make Ferdi because of a fullscreen similar to your last screenshot 😛 It's too intrusive this way, that's why I suggested to add it in the "Support Ferdi" section.

> We could have the link to Typeform there but I am not sure how many **click-through**, and responses, we would get. I am afraid the **volume** would be too low. Realistically, we need to collect **300-350 answers over a period of a few days**. If you think that could work, we can try to have a link/button to Typeform on that screen.

What if we had no expectations? 😄 Let's integrate the Typeform so there is a funnel for user to give their feedback following the methodology you described, but let's not focus on the volume; let's focus on providing Ferdi's value* to everyone and the feedback (via the support section) will follow.

> Oh okay, are we talking about a webview here?

No, no, a pop-up (new window), without an address bar and properly sized so there is not all this white gap we can see in the Typeform in your screenshot. That would be less intrusive and give a clear intent to the pop-up / new window.

**Comment by vantezzen** [comment]
I'd suggest creating a small banner at the bottom of Ferdi, similar to the one you get when there is an update available. This way it wouldn't be too intrusive but also not too hidden away in the support screen.

**Comment by yourcontact** [comment]
> @yourcontact We originally forked Franz to make Ferdi because of a similar fullscreen 😛 It's too intrusive this way, that's why I suggested to add it in the "Support Ferdi" section.

Cool. That is what I thought in the first place.

> What if we had no expectations? 😄 Let's integrate the Typeform so there is a funnel for users to give their feedback following the methodology you described, but let's not focus on the volume; let's focus on providing Ferdi's value to everyone (through better features and increased stability/maintainability 🙏) and the feedback (via the support section) will follow.

That makes sense. I imagine we could work on this as part of #651 Thanks for clarifying over our call.

### Pop-up or browser

>>Oh okay, are we talking about a webview here?
>>
> Think more like a pop-up (new window), without an address bar and properly sized so there is not all this white gap we can see in the Typeform in your screenshot. That would be less intrusive and give a clear intent to the pop-up / new window (triggered by a click from the user, not imposed

That would be great. I'd be up for the pop-up option to display the Typeform survey to users who are willing to answer. As discussed with @kytwb over our call, next is to decide where we want to implement the button, or banner, to trigger that pop-up. We don't want to be intrusive. 

> I'd also suggest we simply open the survey in their normal browser. This way it won't obstruct the view to their services and they can come back to it whenever they have the time to.

It seems that @kytwb's suggestion of a pop-up would address the problem of obstruction. I like the minimalistic option of a tailored pop-up window with no address bar. That said, I am okay with either: pop-up from Ferdi, or the user's default browser.


### Banner

As @vantezzen suggested:

> I'd suggest creating a small banner at the bottom of Ferdi, similar to the one you get when there is an update available. This way it wouldn't be too intrusive but also not too hidden away in the support screen.

I like this. I understood from my chat with @kytwb that we could have this in the next stable update. @vantezzen do you have a screenshot of this banner? I have not seen it yet.

**Comment by vantezzen** [comment]
> do you have a screenshot of this banner?

It could look something like this:

![Screenshot 2020-04-28 at 12 03 42](https://user-images.githubusercontent.com/10333196/80474879-51b9c080-8948-11ea-816f-7b953ebfefe7.png)

**Comment by kytwb** [comment]
@all-contributors please add yourcontact for ideas

**Comment by yourcontact** [comment]
This is an update following our chat with @kytwb 

The next step is to implement the pop-up mechanism as part of the refactoring of the support screen #651. There will also be a survey link on the support screen.

Once we are done with the implementation of the pop-up. We will resume working here to run the survey. We are thinking of introducing the survey link on a [banner](https://github.com/getferdi/ferdi/issues/663#issuecomment-620508372) which will open a pop-up when users click on it.

**Comment by vantezzen** [comment]
Probably not 100% related to this, but I just wanted to inform you that I now started to collect some statistics about Ferdi's release download stats over time:
I have a cronjob running that collects the number of downloads for Windows, mac and Linux (excluding metadata files like "latest.yml") every hour. This data is saved as a simply CSV table.
This data could be used to create some charts on what action brought how many new downloads etc.

If anyone is interested in using this data, here is a small set of the data from the last 18 hours - we may need to look into how we can save and distribute this dataset best:

https://pastebin.com/kKHTf3EE

**Comment by kytwb** [comment]
I think this is very interesting: https://thebrowser.company/

**Comment by vraravam** [comment]
Closing since there's no traction for > 1 year
